### ISSUE TITLE
Pure Parser

### ISSUE THREAD

**Issue by StevenXL** [NorfairKing/yamlparse-applicative#5]
We should expose a way to "read a config file" in a *pure* way in order to test the parser. Is this already implemented and I am just missing it?

At a minimum, I am thinking of something along the lines of:

```haskell
readConfig :: (YamlSchema a, FromJSON a) => ByteString -> Maybe a
```

[readConfigFile](https://hackage.haskell.org/package/yamlparse-applicative-0.1.0.1/docs/YamlParse-Applicative.html#v:readConfigFile) could then be implemented in terms of `readConfig`.

**Comment by NorfairKing** [comment]
I agree that there should be a way to do this! We can use https://hackage.haskell.org/package/yaml-0.11.4.0/docs/Data-Yaml.html#v:decodeEither-39-

**Comment by StevenXL** [comment]
> I agree that there should be a way to do this! We can use https://hackage.haskell.org/package/yaml-0.11.4.0/docs/Data-Yaml.html#v:decodeEither-39-

Exactly what I was thinking. Right now pure testing of the parser an be achieved by digging into the `yamlparse-applicative` dependencies by:

1. Having a Yaml encoded bytestring - e.g. `let yamlBS = "token: '123'\n"`
2. Getting a value of type `Data.Aeson.Value` - e.g. `value <- Data.Yaml.decodeThrow yamlBS`
3. Getting a value of type `Data.Aeson.Parser` - e.g., `viaYamlSchema value`
4. Once you have 3, you can use the many different ways to run an Aeson parser such a`https://hackage.haskell.org/package/aeson-1.4.7.1/docs/Data-Aeson-Types.html#v:parse`. 

I'd like to continue working on `hastory`, but once I pause on there to get your feedback, I'll circle back to this.
